### DISCUSSION TITLE
Use index.html in public/ folder as directory index

### DISCUSSION THREAD

**Discussion by Vinnl** [vercel/next.js#14361] [Help]
# Feature request

## Is your feature request related to a problem? Please describe.

I'm bundling a static website (specifically, a built [Storybook](https://storybook.js.org/)) into the `public/` folder of my Next.js app. That means that there is a file `/public/storybook/index.html`.

Before migrating to Next.js, I could visit `mywebsite.com/storybook/`. Now, however, I can only get it to show up properly when visiting `mywebsite.com/storybook/index.html`.

(This is when using `next start`, not on Vercel. Sorry!)

## Describe the solution you'd like

For the `index.html` file to be able to serve as the directory index for the folder it's in, if it's located within the `public/` folder (i.e. what happens when `index: true` for express.static).

## Describe alternatives you've considered

Just visiting `index.html`, or somehow integrating Storybook with my Next.js app.

## Additional context

I really hope this hasn't been reported before, but I couldn't find anything with these search terms:

- index.html directory index 
- index.html static
- index.html public
- index.html

**Comment by tdurand** [comment]
I'm interested in doing this also  .. I'm trying to play with `vercel.json` to get the same behaviour as `express.static` but can't find a way

**Comment by spidgorny** [comment]
I've checked the source for [generatePublicRoutes()](https://github.com/vercel/next.js/blob/9dd5ff2baa716a6b12f681ff09559a3c8dd7b5cd/packages/next/next-server/server/next-server.ts#L1004)
```
const publicFiles = new Set(
      recursiveReadDirSync(this.publicDir).map((p) =>
        encodeURI(p.replace(/\\/g, '/'))
      )
    )
```
and later:
```
if (publicFiles.has(path)) {
  // ok, render existing file
}
```

So it only works for real file requests like ```/storybook/index.html```. I'd be glad to implement a PR, just need some guidance.

**Comment by ghost** [comment]
I'm testing out the following in my next config:
```
  async rewrites() {
    return {
     fallback: [
       {
         source: '/:path*',
         destination: '/:path*/index.html',
       },
     ],
   }
  },
```

In my case, I want local development to match the behavior of Vercel deployments. I currently observe Vercel deployments applying this rewrite. There is no documentation that indicates that this rewrite rule exists on Vercel.
